onadata/apps/api/viewsets/xform_viewset.py

```diff
@@ -337,17 +337,21 @@ def create_async(self, request, *args, **kwargs):
                                 status=status.HTTP_400_BAD_REQUEST)
 
             fname = request.FILES.get('xls_file').name
+            if isinstance(request.FILES.get('xls_file'), InMemoryUploadedFile):
+                xls_file_path = default_storage.save(
+                        f'tmp/async-upload-{owner.username}-{fname}',
+                        ContentFile(request.FILES.get('xls_file').read()))
+            else:
+                xls_file_path = request.FILES.get(
+                    'xls_file').temporary_file_path()
+
             resp.update(
                 {u'job_uuid':
                  tasks.publish_xlsform_async.delay(
-                     request.user, request.POST, owner,
-                     ({'name': fname,
-                       'data': request.FILES.get('xls_file').read()}
-                      if isinstance(request.FILES.get('xls_file'),
-                                    InMemoryUploadedFile) else
-                      {'name': fname,
-                       'path': request.FILES.get(
-                           'xls_file').temporary_file_path()})).task_id})
+                     request.user.id,
+                     request.POST,
+                     owner.id,
+                     {'name': fname, 'path': xls_file_path}).task_id})
```

Comment:
Had to modify this since celery tries to serialize the passed in arguments and fails. Not sure when the serialization of arguments was introduced... but passing in primitive / serializable data was the solution I thought would work here.